crates/matrix-sdk-base/src/client.rs

```diff
@@ -912,27 +884,39 @@ impl BaseClient {
             let (raw_state_events, state_events) =
                 processors::state_events::collect_sync(&mut context, &new_info.state.events);
 
-            let mut user_ids = processors::state_events::dispatch_and_get_new_users(
+            let mut new_user_ids = processors::state_events::dispatch_and_get_new_users(
                 &mut context,
                 (&raw_state_events, &state_events),
                 &mut room_info,
                 &mut ambiguity_cache,
             )
             .await?;
 
+            let (raw_state_events_from_timeline, state_events_from_timeline) =
+                processors::state_events::collect_sync_from_timeline(
+                    &mut context,
+                    &new_info.timeline.events,
+                );
+
+            let mut other_new_user_ids = processors::state_events::dispatch_and_get_new_users(
+                &mut context,
+                (&raw_state_events_from_timeline, &state_events_from_timeline),
+                &mut room_info,
+                &mut ambiguity_cache,
+            )
+            .await?;
+            new_user_ids.append(&mut other_new_user_ids);
```

Comment:
Same. Because `handle_timeline` doesn't handle state events, let's do it here, prior the call to `handle_timeline`.